docs/drive-replacement.md

```diff
@@ -0,0 +1,237 @@
+Drive replacement (applies for versions > v3.2.2)
+-------------
+
+These are the steps to move the volumes from one drive to another drive from the **same node**. These steps will be useful during drive replacement.
+
+(NOTE: This will just move the volume references from source to the destination drive without moving the actual data)
+
+## STEP-BY-STEP Guide for drive replacement
+
+### STEP 1: Add new drive
+
+Once the new drive has been attached to the node as a replacement of an older drive, Discover and initialize the newly added drive by executing the following steps
+
+```sh
+$ kubectl directpv discover --drives /dev/<drive-name>
+```
+
+This command will generate `drives.yaml` which will be used later for initialization. Please refer [Discover CLI](./cli.md#discover-drives) for more helpers on discovery.
+
+Now, initialize the drive using the `drives.yaml` file that was created by the previous command
+
+```sh
+$ kubectl directpv init drives.yaml
+```
+
+On success, you should see the new drive listed in `kubectl directpv list drives --drives /dev/<drive-name>`
+
+### STEP 2: Get source and destination drive IDs
+
+```sh
+$ kubectl directpv list drives --drives <source-drive-name>,<dest-drive-name> --nodes <node-name> -o wide
+```
+
+### STEP 3: Cordon source and destination drives
+
+By cordoning, we mark the drives as unschedulable so that no new volumes will be scheduled on these drives
+
+```sh
+$ kubectl directpv cordon <source-drive-id> <dest-drive-id>
+```
+
+### STEP 4: Cordon the kubernetes node and delete the corresponding workload pod that is running on this node
+
+Cordon the corresponding k8s node and delete the corresponding wokload pod, this will take the workloads offline and also doesn't schedule any new pods on this node.
+
+```sh
+$ kubectl cordon <node-name>
+```
+
+### STEP 5: Get the pods used by the source drive
+
+Make a note of the pods to be deleted in their respective namespaces
+
+```sh
+$ kubectl directpv volumes list --all --drives /dev/<source-drive-name> --nodes <node-name>
+```
+
+### STEP 6: Delete the pods used by the source drive
+
+(NOTE: Avoid force deleting here as force deletion might not unpublish the volumes)
+
+```sh
+$ kubectl delete pod/<pods-name> -n <pod-ns>
+```
```

Comment:
Step 4 calls for deleting worker pods already - does cordoning already do that? Or should we drop that from step 4 and just have it down here.